### PR TITLE
10741 nil carto visualization check for nm update

### LINKED ISSUE #10741: Named map update failing due to nil `carto_visualization`

### PR DESCRIPTION
**Pull request by gfiorav**
Fixes #10741

Please CR @javitonino

### CODE REVIEW

File: app/models/visualization/member.rb [hunk 1]
```diff
@@ -838,13 +838,19 @@ def create_named_map
...
+
+        # A visualization destroy triggers destroys on all its layers. Each
+        # layer destroy, will trigger an update back to the visualization. When
+        # the last layer is destroyed, and the visualization named map template
+        # is generated to be updated, it will contain no layers, causing an
+        # error at the Maps API. This is a hack to prevent that update and error
+        # from happening. A better way to solve this would be to get
+        # callbacks under control.
+        if presentation && presentation.layers.any?
+          Carto::NamedMaps::Api.new(presentation_visualization).update
```
Comment (javitonino) [inline app/models/visualization/member.rb]: `presentation_visualization` looks wrong. Did you mean `presentation`?

File: app/models/visualization/member.rb [hunk 2]
```diff
@@ -838,13 +838,19 @@ def create_named_map
...
         return if named_map_updates_disabled?
 
-        # On visualization destroy, an update will be performed after every
-        # layer in the vis is destroyed until the template has no layers
-        # and the update fails. This is a hacky way to fix that. A better way
-        # would be to fix callbacks.
-        visualization_for_presentation = carto_visualization.for_presentation
-        unless visualization_for_presentation.layers.empty?
-          Carto::NamedMaps::Api.new(carto_visualization.for_presentation).update
+        presentation = if carto_visualization
```
Comment (javitonino) [inline app/models/visualization/member.rb]: This could be a one liner with `try`, easier to read that the 3-lines block I think.

**Comment by gfiorav** [comment]
Carless push, stay tuned.

**Comment by gfiorav** [comment]
See now, sorry for previous ping

**Pull request merged**